src/chains/ethereum/ethereum/src/miner/miner.ts

```diff
@@ -424,13 +423,7 @@ export default class Miner extends Emittery<{
         storageKeys
       );
       block = finalizedBlockData.block;
-      const emitBlockProm = this.emit("block", finalizedBlockData);
-      if (legacyInstamine === true) {
-        // we need to wait for each block to be done mining when in legacy
-        // mode because things like `mine` and `miner_start` must wait for the
-        // first mine operation to be fully complete.
-        await emitBlockProm;
-      }
+      this.emit("block", finalizedBlockData);
```

Comment:
We don't need to wait because `evm_mine` and `miner_start` wait on their own.

Reply:
If you feel it necessary to explain this in the review it likely belongs as a comment in the code.

Reply:
It explains why the old code was removed.

Reply:
I'm actually having to revisit this because I noticed if someone sends stops the miner via miner_stop, send enough transactions to fill two blocks, then calls `miner_start`, all pending transactions aren't mined before `miner_start` returns... which is what we want.

I think `evm_mine` suffers from a similar issue, but I haven't yet figured out what yet.

These issues might be rare enough that I can wait until after v7 stable to fix them, as it might take a bit of refactoring to get the timings correct.